### ISSUE TITLE
🐞 [Bug]: The GPU endpoint for the node returns a value that can be null.

### ISSUE THREAD

**Issue by Mahmoud-Emad** [threefoldtech/tfgrid-sdk-go#1051]
### What happened?

 I rented more than one node, and all worked fine as I could view the GPU details of the nodes. However, I encountered a problem with one node as I could not list its GPU. After investigating, I discovered that the node's GPU returned a `null` value from the grid proxy.

This is a [valid node](https://gridproxy.grid.tf/nodes/14/) and this is its [GPU](https://gridproxy.grid.tf/nodes/14/gpu).
This is a [not valid node](https://gridproxy.grid.tf/nodes/5856) and this is its [GPU](https://gridproxy.grid.tf/nodes/5856/gpu).

### which network/s did you face the problem on?

Main

### Twin ID/s

8161

### Version

2.4.2

### Node ID/s

5856

### Farm ID/s

1840

### Relevant log output

```console
N/A
```

**Comment by Omarabdul3ziz** [comment]
node 5856 doesn't have a gpu but its num_gpu=1, check the triggers

**Comment by Omarabdul3ziz** [comment]
> tl;dr issue fixed in `v0.15.0` release

here is what happened.
- checking the processor_db on the main grid stack found that the node_gpu table already has a GPU for this node twin 
  ```
  tfgrid-graphql=# select * from node_gpu where node_twin_id=9545;
   node_twin_id |           id           |       vendor       |             device             | contract 
  --------------+------------------------+--------------------+--------------------------------+----------
          9545 | 0000:02:00.0/10de/1245 | NVIDIA Corporation | GF116 [GeForce GTS 450 Rev. 2] |        0
  (1 row)
  ```
  and in the cache table, it is reported correctly
  ```
  tfgrid-graphql=# select node_gpu_count from resources_cache where node_id=5856;
   node_gpu_count 
   ----------------
                1
  (1 row)
  ```
  so it is not a problem with the triggers.

- the most reasonable scenario could have happened is that the node had a GPU once but it is not there anymore. there is an invalidation mechanism introduced here https://github.com/threefoldtech/tfgrid-sdk-go/pull/642 for a similar issue. but it wasn't working as expected so it was modified later in this pr https://github.com/threefoldtech/tfgrid-sdk-go/pull/726
- checking the node_gpu table it doesn't have an updated_at field which is what the new invalidation mechanism depends on to consider the GPU card as expired. this means this proxy doesn't have the latest changes regarding this part yet.
- checking the [changelog](https://github.com/threefoldtech/tfgrid-sdk-go/blob/development/grid-proxy/CHANGELOG.md) the latest expiration should be in release `v0.15.0` but the main proxy still ar `v0.14.13`

**Comment by Omarabdul3ziz** [comment]
this should be verified carefully on mainnet

**Comment by ashraffouda** [comment]
already fixed in a previous  0.15.0  which is not on mainnet yet
